sys/net/gnrc/network_layer/ipv6/nib/_nib-internal.c

```diff
@@ -0,0 +1,410 @@
+/*
+ * Copyright (C) 2016 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ * @author  Martine Lenders <m.lenders@fu-berlin.de>
+ */
+
+#include <stdbool.h>
+#include <string.h>
+
+#include "net/gnrc/ipv6.h"
+#include "net/gnrc/ipv6/nib/conf.h"
+#include "net/gnrc/ipv6/nib/nc.h"
+#include "net/gnrc/ipv6/nib.h"
+#include "net/gnrc/netif.h"
+#include "random.h"
+
+#include "_nib-internal.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+/* pointers for default router selection */
+static _nib_dr_t *_prime_def_router = NULL;
+static _nib_dr_t *_last_def_router = NULL;
+static clist_node_t _next_removable = { NULL };
+
+static _nib_t _nodes[GNRC_IPV6_NIB_NUMOF];
+static _nib_dr_t _def_routers[GNRC_IPV6_NIB_DEFAULT_ROUTER_NUMOF];
+static _nib_iface_t _nis[GNRC_NETIF_NUMOF];
+
+#if ENABLE_DEBUG
+static char addr_str[IPV6_ADDR_MAX_STR_LEN];
+#endif
+
+mutex_t _nib_mutex = MUTEX_INIT;
+
+static void _override_node(const ipv6_addr_t *addr, unsigned iface,
+                           _nib_t *node);
+static inline bool _node_unreachable(_nib_t *node);
+
+void _nib_init(void)
```

Comment:
Yes, but for the testing case it is called for every test. Also note the TODO below, which is a *required* initialization step [via the RFC](https://tools.ietf.org/html/rfc6775#section-8.1.1). We can however set the zeroing into `#ifdef TEST_SUITES`, if that is preferred.